dev/README_AIRFLOW3_DEV.md

```diff
@@ -0,0 +1,85 @@
+<!--
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+-->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of contents**
+
+- [Main branch is Airflow 3](#main-branch-is-airflow-3)
+- [Contributors](#contributors)
+  - [Developing for providers and Helm chart](#developing-for-providers-and-helm-chart)
+  - [Developing for Airflow 3](#developing-for-airflow-3)
+  - [Developing for Airflow 2.10.x](#developing-for-airflow-210x)
+  - [Developing for Airflow 3 and 2.10.x / 2.11.x](#developing-for-airflow-3-and-210x--211x)
+- [Committers / PMCs](#committers--pmcs)
+  - [Merging PRs 2.10.x](#merging-prs-210x)
+  - [Merging PRs for Airflow 3](#merging-prs-for-airflow-3)
+  - [Merging PR for Airflow 3 and 2.10.x / 2.11.x](#merging-pr-for-airflow-3-and-210x--211x)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Main branch is Airflow 3
+
+The main branch is for development of Airflow 3.
+Airflow 2.10.x releases will be cut from `v-2-10-stable` branch.
+
+# Contributors
+
+The following section explains which branch you should target your PR
+
+## Developing for providers and Helm chart
+
+PRs should target `main` branch.
+
+## Developing for Airflow 3
+
+PRs should target `main` branch.
+
+## Developing for Airflow 2.10.x
+
+PR should target `v-2-10-test` branch. When cutting a new release for 2.10 release manager
+will sync `v-2-10-test`  branch to `v-2-10-stable` and cut the release from the stable branch.
+PR should never target `v-2-10-stable` unless specifically instructed by release manager.
+
+## Developing for Airflow 3 and 2.10.x / 2.11.x
+
+If the PR is relevant for both Airflow 3 and 2 it should target `main` branch.
```

Comment:
```suggestion
If the PR is relevant for both Airflow 3 and 2 it should target `main` branch.
It is the responsibility of the committer to ensure it will be back-ported to `v-2-10-test` maintenance branch (but help by contributor is appreciated).
```

Reply:
Note, I'd favor to bring this chapter before the "Developing for Airflow 2.10.x" as this should be favored compared to only contributing to 2.10.x line in my view. Especially bugfixes.